### PR TITLE
Turn deprecation warning into error: invalid toolchain spec

### PR DESCRIPTION
**Pull request by alllex**
Fixes https://github.com/gradle/gradle/issues/21794

### CODE REVIEW

**Comment by alllex** [comment]
@bot-gradle test this

**Comment by alllex** [comment]
@bot-gradle test this

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc [hunk 1]
```diff
@@ -193,7 +193,7 @@ systemProp.org.gradle.kotlin.dsl.precompiled.accessors.strict=true
 
 Along with the Java language version, the <<toolchains#toolchains, Java toolchain>> DSL allows configuring other criteria such as specific vendors or VM implementations.
 Starting with Gradle 7.6, toolchain specifications that configure other properties without specifying the language version are considered _invalid_.
-Invalid specifications are deprecated and will become build errors in Gradle 8.0.
+Usage of invalid specifications results in build errors since Gradle 8.0.
```
Comment (ljacomet) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc:196]: I do not think this entry should be updated.
Instead you should add a new entry to the 8.0 items "Warnings that are now errors"

**Review by ljacomet — CHANGES_REQUESTED** [review]
Looks good overall, only comment is about documentation.

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc (hunk 1 above)
Comment (alllex) [reply subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc]: Done

**Review by ljacomet — APPROVED** [review]
LGTM

**Comment by alllex** [comment]
@bot-gradle test and merge

**Pull request merged**